src/System.Management.Automation/FormatAndOutput/common/Utilities/MshObjectUtil.cs

```diff
@@ -290,10 +296,12 @@ internal static string SmartToString(PSObject so, PSPropertyExpressionFactory ex
                 // take care of the case there is no base object
                 return so.ToString();
             }
-            catch (ExtendedTypeSystemException e)
+            catch (Exception e) when (e is ExtendedTypeSystemException || e is InvalidOperationException)
             {
-                // NOTE: we catch all the exceptions, since we do not know
-                // what the underlying object access would throw
+                // These exceptions are being caught and handled by returning an empty string when
+                // the object cannot be stringified due to ETS or an instance in the collection has been modified
+                Tracer.WriteLine("Exception during conversion to string, emitting empty string.");
```

Comment:
Maybe TraceWarning()?
```suggestion
                s_tracer.TraceWarning("SmartToString method: Exception during conversion to string, emitting empty string.");
```